modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md

```diff
@@ -0,0 +1,135 @@
+- пытаемся запустить etcd
+- если ошибка выше ушла, проверяем статус
+```bash
+ETCDCTL_API=3 etcdctl --cacert /etc/kubernetes/pki/etcd/ca.crt --cert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/ca.key --endpoints https://127.0.0.1:2379/ endpoint status -w table
+```
+- если имеем ошибку `alarm:NOSPACE`, выполняем:
+```bash
+ETCDCTL_API=3 etcdctl --cacert /etc/kubernetes/pki/etcd/ca.crt --cert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/ca.key --endpoints https://127.0.0.1:2379/ alarm disarm
+```
+- дефрагментируем etcd если это необходимо:
+```bash
+ETCDCTL_API=3 etcdctl --cacert /etc/kubernetes/pki/etcd/ca.crt --cert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/ca.key --endpoints https://127.0.0.1:2379/ defrag --command-timeout=60s
+```
+
+### Восстанавливаемся из бекапа
+- загружаем на сервер [etcdctl](https://github.com/etcd-io/etcd/releases), желательно той же версии, что и на сервере
+- останавливаем etcd (перемещаем манифест пода)
+- удаляем папку c данными `rm -rf /var/lib/etcd/`
+- восстанавливаем базу `ETCDCTL_API=3 etcdctl snapshot restore /var/lib/etcd-backup/member/snap/db --cacert /etc/kubernetes/pki/etcd/ca.crt --cert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/ca.key --endpoints https://127.0.0.1:2379/  --data-dir=/var/lib/etcd --skip-hash-check`
+- пытаемся запустить etcd
+
+## Multi-master
+
+### Потеря кворума
+- добавляем в манифест аргумент `--force-new-cluster` на оставшейся ноде
+- ждем поднятия etcd
+- удаляем аргумент `--force-new-cluster`
+Далее, если ноды были потеряны безвозвратно, то добавлям новые через `dhctl converge` или вручную, если кластер статический.
+
+Если ноды возвратились, то они уже не являются членами кластера.
+На узлах, которые не являются членами кластера выполняем следующие операции:
+- останавливаем etcd (перемещаем манифест пода)
+- бекапим файлы (на всякий случай)
+- удаляем папку c данными `rm -rf /var/lib/etcd/`
+
+Возвращаемся на рабочий кластер, и перезапускаем DaemonsSet `d8-control-plane-manager`:
+```bash
+kubectl -n kube-system rollout restart daemonset d8-control-plane-manager
+```
+Ожидаем переката всех Pod'ов control-plane и переход их в состояние `Ready` и смотрим что все etcd инстансы стали участниками кластера.
+```
+ETCDCTL_API=3 etcdctl --cacert /etc/kubernetes/pki/etcd/ca.crt --cert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/ca.key --endpoints https://127.0.0.1:2379/ member list -w table
+```
+### Полная потеря данных и восстановление из бекапа
+
+На всех нодах:
+- останавливаем etcd (перемещаем манифест пода)
+- выполняем резервное копирование файлов (на всякий случай)
+- удаляем папку c данными `rm -rf /var/lib/etcd/`
+- восстанавливаем на одной ноде как для [single-мастера](#восстанавливаемся-из-бекапа)
```

Comment:
Don't we need to make single-master first if we have multi-master?

Reply:
yep, i added it